plugin-desktopswitch/desktopswitch.cpp

```diff
@@ -219,9 +224,27 @@ void DesktopSwitch::onNumberOfDesktopsChanged(int count)
 
 void DesktopSwitch::onCurrentDesktopChanged(int current)
 {
-    QAbstractButton *button = m_buttons->button(current - 1);
-    if (button)
-        button->setChecked(true);
+    if (mShowOnlyActive)
+    {
+        int i = 1;
+        for (auto button : m_buttons->buttons())
```

Comment:
Possible container detachment. ```qAsConst(m_buttons->buttons())```

Reply:
There is no need to cast here, because the returned list is a temporary object.
Actually you can't use qAsConst here -> https://doc.qt.io/qt-5/qtglobal.html#qAsConst-1

Reply:
rvalue overload is deleted. I know that. nevermind that comment.
A non const rvalue can still detach, tough.